src/org/kms/core/MKRequester.java

```diff
@@ -60,6 +58,12 @@ public String getMasterKeyForVersion(String encryptedMKStr){
 	return getMasterKey();
 }
 
+private String getUserIdFromRequestMK(){
+	int loc = mkObjectName.indexOf("_");
```

Comment:
I create the mkObjectName and it always will contain '_'. That is why i haven't added a check. But since it is a good practice to check for negative scenarios, I will add it.